anchor-service/src/anchor.rs

```diff
@@ -76,15 +100,123 @@ pub struct TimeEventBatch {
     /// The anchor proof
     pub proof: Proof,
     /// The Time Events
-    pub time_events: TimeEvents,
+    pub raw_time_events: RawTimeEvents,
 }
 
 impl std::fmt::Debug for TimeEventBatch {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("TimeEventBatch")
             .field("merkle_nodes", &self.merkle_nodes)
             .field("proof", &self.proof)
-            .field("time_events", &self.time_events)
+            .field("raw_time_events", &self.raw_time_events)
             .finish()
     }
 }
+
+impl TimeEventBatch {
+    /// Try to convert the TimeEventBatch into a list of TimeEventInsertables
+    pub fn try_to_insertables(&self) -> Result<Vec<TimeEventInsertable>> {
```

Comment:
maybe this can return a stream rather than collect the entire set into a vector in memory? I still need to look where/how it's used but just off the top of my head

Reply:
Who would own the TimeEventInsertables in that case.

Reply:
yeah, this is fine. I thought more about it and we're already reading everything from the db in one shot so there's no point in streaming anywhere until we stream that